tests/JoinArrayTest.php

```diff
@@ -44,6 +44,7 @@ public function testDirection()
         $this->assertSame('test_id', $this->getProtected($j, 'master_field'));
         $this->assertSame('id', $this->getProtected($j, 'foreign_field'));
 
+        $this->expectException(Exception::class);
```

Comment:
move `expectException` as close as possible to the statement which throw, if the next statement  throws, delete unneeded assertions